### PR TITLE
test: Enable Comet shuffle in Spark SQL tests

### LINKED ISSUE #195: Enable shuffle in Spark SQL tests
### What is the problem the feature request solves?

In https://github.com/apache/arrow-datafusion-comet/pull/166 we only enabled Spark SQL tests without Comet shuffle. We should enable it later.

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sunchao**
## Which issue does this PR close?

Closes #195.

## Rationale for this change

Currently the Spark SQL tests do not enable Comet shuffle yet, which causes it to miss quite a lot test coverage. This PR enables shuffle for them. It uses the default shuffle mechanism for now (Comet native shuffle).

I have to disable a few tests that are failing probably due to some bugs in Comet itself. Once this is merged, I'll create separate Github issues to track them.

## What changes are included in this PR?

Enable Comet shuffle for Spark SQL tests.

### CODE REVIEW

**Comment by sunchao** [comment]
cc @viirya @advancedxy this PR is ready now

File: dev/diffs/3.4.2.diff [hunk 1]
```diff
@@ -1238,6 +1392,9 @@ index ed2e309fa07..3767d4e7ca4 100644
 +        conf
 +          .set("spark.comet.exec.enabled", "true")
 +          .set("spark.comet.exec.all.enabled", "true")
++          .set("spark.shuffle.manager",
++            "org.apache.spark.sql.comet.execution.shuffle.CometShuffleManager")
++          .set("spark.comet.exec.shuffle.enabled", "true")
```
[the remark this answers is not in the document]
Comment (sunchao) [reply dev/diffs/3.4.2.diff:1397]: It is still pending on the work to make it as default. Once it becomes default, all the tests will automatically switch to use it.

File: dev/diffs/3.4.2.diff [hunk 2]
```diff
@@ -234,6 +234,20 @@ index 56e9520fdab..917932336df 100644
...
+index 9ddb4abe98b..1bebe99f1cc 100644
+--- a/sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala
++++ b/sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala
+@@ -3311,7 +3311,8 @@ class DataFrameSuite extends QueryTest
+     assert(df2.isLocal)
+   }
+ 
+-  test("SPARK-35886: PromotePrecision should be subexpr replaced") {
++  test("SPARK-35886: PromotePrecision should be subexpr replaced",
++    IgnoreComet("TODO: fix Comet for this test")) {
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:247]: This will be forgotten easily. Can we create ticket for that?

File: dev/diffs/3.4.2.diff [hunk 3]
```diff
@@ -1139,6 +1205,94 @@ index 75f440caefc..36b1146bc3a 100644
...
++import org.apache.spark.sql.IgnoreComet
+ import org.apache.spark.sql.catalyst.streaming.InternalOutputModes.Update
+ import org.apache.spark.sql.execution.streaming.{FlatMapGroupsWithStateExec, MemoryStream}
+ import org.apache.spark.sql.internal.SQLConf
+@@ -91,7 +92,7 @@ class FlatMapGroupsWithStateDistributionSuite extends StreamTest
+   }
+ 
+   test("SPARK-38204: flatMapGroupsWithState should require StatefulOpClusteredDistribution " +
+-    "from children - without initial state") {
++    "from children - without initial state", IgnoreComet("TODO: fix Comet for this test")) {
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:1225]: And this.

File: dev/diffs/3.4.2.diff [hunk 4]
```diff
@@ -1139,6 +1205,94 @@ index 75f440caefc..36b1146bc3a 100644
...
+     // function will return -1 on timeout and returns count of the state otherwise
+     val stateFunc =
+       (key: (String, String), values: Iterator[(String, String, Long)],
+@@ -243,7 +244,8 @@ class FlatMapGroupsWithStateDistributionSuite extends StreamTest
+   }
+ 
+   test("SPARK-38204: flatMapGroupsWithState should require ClusteredDistribution " +
+-    "from children if the query starts from checkpoint in 3.2.x - without initial state") {
++    "from children if the query starts from checkpoint in 3.2.x - without initial state",
++    IgnoreComet("TODO: fix Comet for this test")) {
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:1235]: ditto

File: dev/diffs/3.4.2.diff [hunk 5]
```diff
@@ -1139,6 +1205,94 @@ index 75f440caefc..36b1146bc3a 100644
...
+     // function will return -1 on timeout and returns count of the state otherwise
+     val stateFunc =
+       (key: (String, String), values: Iterator[(String, String, Long)],
+@@ -335,7 +337,8 @@ class FlatMapGroupsWithStateDistributionSuite extends StreamTest
+   }
+ 
+   test("SPARK-38204: flatMapGroupsWithState should require ClusteredDistribution " +
+-    "from children if the query starts from checkpoint in prior to 3.2") {
++    "from children if the query starts from checkpoint in prior to 3.2",
++    IgnoreComet("TODO: fix Comet for this test")) {
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:1245]: ditto.

File: dev/diffs/3.4.2.diff [hunk 6]
```diff
@@ -1139,6 +1205,94 @@ index 75f440caefc..36b1146bc3a 100644
...
+-import org.apache.spark.sql.{DataFrame, Encoder}
++import org.apache.spark.sql.{DataFrame, Encoder, IgnoreCometSuite}
+ import org.apache.spark.sql.catalyst.InternalRow
+ import org.apache.spark.sql.catalyst.expressions.{GenericInternalRow, UnsafeProjection, UnsafeRow}
+ import org.apache.spark.sql.catalyst.plans.logical.FlatMapGroupsWithState
+@@ -46,8 +46,9 @@ case class RunningCount(count: Long)
+ 
+ case class Result(key: Long, count: Int)
+ 
++// TODO: fix Comet to enable this suite
```
Comment (viirya) [inline dev/diffs/3.4.2.diff:1266]: Hmm, all this suite fail?

**Review by viirya — APPROVED** [review]
Looks good to me. I suggested that we can create tickets for the failed tests.

**Comment by viirya** [comment]
I created #231 to track these failed tests.

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request merged**